diag_manager/fms_diag_field_object.F90

```diff
@@ -1235,8 +1237,10 @@ subroutine write_field_metadata(this, fms2io_fileobj, file_id, yaml_id, diag_axi
   endif
 
   if (this%has_interp_method()) then
-    call register_variable_attribute(fms2io_fileobj, var_name, "interp_method", this%get_interp_method(), &
-      str_len=len_trim(this%get_interp_method()))
+    interp_method_tmp = this%interp_method
```

Comment:
Is this to deal with some compiler peculiarity?

Reply:
Yeah the previous code triggers an internal compiler error with gcc